src/lib/lore.sh

```diff
@@ -458,3 +464,152 @@ function download_series()
   done
   wait
 }
+
+# This function deletes a patch series from the local storage
+#
+# @download_dir_path: The path to the directory where the series was stored
+# @patch_title: The title of the patch
+# @flag: Flag to control function output
+function delete_series_from_local_storage()
+{
+  local download_dir_path="$1"
+  local patch_title="$2"
+  local flag="$3"
+  local patch_file_name
+  local file_pattern
+  local command
+
+  flag=${flag:-'SILENT'}
+
+  patch_file_name=$(convert_title_to_patch_name "$patch_title")
+
+  file_pattern=$(printf '%s/*%s' "${download_dir_path}" "${patch_file_name}")
+
+  ls "${file_pattern}" 1> /dev/null 2>&1
+
+  if [[ "$ret" == 0 ]]; then
+    command="rm ${file_pattern}"
+    cmd_manager "$flag" "$command"
+  else
+    return 2 # ENOENT
+  fi
+}
+
+# This function creates the lore bookmarked series file if it doesn't
+# already exists
+function create_lore_bookmarked_file()
+{
+  if [[ ! -f "${BOOKMARKED_SERIES_PATH}" ]]; then
+    touch "${BOOKMARKED_SERIES_PATH}"
+  fi
+}
+
+# This function adds a given series to a local bookmark database for
+# series managed by kw. To achieve uniqueness for each series the function
+# assigns an id to it. Also a timestamp for informational purposes is also
+# added.
+#
+# @target_patch: Patch (series) in the same format as list_of_mailinglist_patches
+#   to be added to local bookmark database
+# @download_dir_path: The directory where the patch (series) was saved
+function add_series_to_bookmark()
+{
+  local target_patch="$1"
+  local download_dir_path="$2"
+  local patch_id
+  local timestamp
+
+  create_lore_bookmarked_file
+
+  patch_id=$(printf '%s' "${target_patch}" | sha256sum | cut -d ' ' -f1)
+  timestamp=$(date '+%Y/%m/%d %H:%M')
+
+  if [[ $(grep -c "^${patch_id}" "${BOOKMARKED_SERIES_PATH}") == 0 ]]; then
```

Comment:
Use `--count` instead of `-c`.
Additionally, I would remove the `grep` command from the `if` condition, and I would just use `"$?" == 0`.